protocol/rpcprovider/rpcprovider_server.go

```diff
@@ -180,7 +183,12 @@ func (rpcps *RPCProviderServer) craftChainMessage() (chainMessage chainlib.Chain
 
 // function used to handle relay requests from a consumer, it is called by a provider_listener by calling RegisterReceiver
 func (rpcps *RPCProviderServer) Relay(ctx context.Context, request *pairingtypes.RelayRequest) (*pairingtypes.RelayReply, error) {
-	grpc.SetTrailer(ctx, metadata.Pairs(chainlib.RpcProviderUniqueIdHeader, rpcps.providerUniqueId))
+	// count the number of simultaneous relay calls
+	rpcps.providerLoadManager.addRelayCall()
+	defer rpcps.providerLoadManager.removeRelayCall()
+	provider_relay_load := strconv.FormatFloat(rpcps.providerLoadManager.getProviderLoad(), 'f', -1, 64)
```

Comment:
also here just call a method from providerLoadManager that returns the metadata already. (empty if providerLoadManager == nil) 

and perform the formatting inside. later just add to the trailer if len(loadMetadata) > 0

Reply:
Fixed in [8b8a05a](https://github.com/lavanet/lava/pull/1720/commits/8b8a05acd54c009763e0aa8de852218165b398f4)